### PR TITLE
Switching between renderers

### PR DESCRIPTION
**Pull request by zgdump**
## Description

Xash3D FWGS engine already has a mechanism for switching renderers via the r_refdll сvar. But now it's broken. The vulkan is not in the list, switching renderers causes glitches, and the window is not recreated.

## Plan

In this PR, I want to fix the switching of renderers, as well as all related problems.

## Behavior changing

The -ref option will override the r_refdll variable and after restarting the render will remain the same as the last time

## Todo

Engine

- [x] Fix r_refdll
- [x] Recreate window if r_reddll changed
- [x] Reinitialize renderer
- [ ] Reload all textures
- [x] Reinit studio models
- [x] Draw map after render changed

Refactoring

- [ ] Call R_Free_Video out of renderer

Settings gui

- [x] Add vulkan renderer into the video modes menu
- [ ] Add vulkan rtx flag into the video modes menu
- [ ] Add note in the video modes settings
- [ ] Do not change render before "apply" clicked

## Refused todos

- Fix fullscreen glitches
- Fix resolution changes

*Thats draft, switch is not working properly!*

### CODE REVIEW

File: engine/common/com_strings.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ GNU General Public License for more details.
 #define XASH_ENGINE_NAME "Xash3D FWGS"
 
 // renderers order is important, software is always a last chance fallback
-#define DEFAULT_RENDERERS { "gl", "gles1", "gles2", "gl4es", "soft" }
```
Comment (w23) [inline engine/common/com_strings.h:69]: Vulkan should be first :sweat_smile:

**Comment by w23** [comment]
Switching between vk and any other renderer requires full window teardown and recreation with different SDL flags. I'm not entirely sure, but I suspect that this would require quite a bit of work on Xash3D engine side.

Can you check that `VID_CreateWindow` gets called when changing renderers?

File: engine/common/com_strings.h (hunk 1 above)
Comment (zgdump) [reply engine/common/com_strings.h:69]: No)

**Comment by zgdump** [comment]
> Switching between vk and any other renderer requires full window teardown and recreation with different SDL flags. I'm not entirely sure, but I suspect that this would require quite a bit of work on Xash3D engine side.
> 
> Can you check that `VID_CreateWindow` gets called when changing renderers?

The advice helped! To switch correctly, we need to recreate the window. With some hacks, it worked. Well... Almost

https://youtu.be/-FhE6jTWogY

**Comment by zgdump** [comment]
@w23 We may need your help and a lot of fixes to switch render correctly

**Comment by zgdump** [comment]
At the moment, switching the renderer is working. But it has three problems that prevent from playing comfortably:
- [ ] Broken textures in menu, console, hud
- [ ] Broken model's textures 
- [ ] Broken textures on levels, until the level change occurs

VK renderer crashes are caused by problems with textures.
It seems that the initialization order is not correct or a function was missed somewhere

**Comment by zgdump** [comment]
Closed. After creating the texture manager, PR will be cleaned and re-opened in the parent repository

**Pull request closed without merging**